lib/valid_email2/address.rb

```diff
@@ -57,7 +58,12 @@ def valid_domain?
     end
 
     def valid_address?
-      return false if address.address != @raw_address
+      if @allow_display_name
+        address_only = @raw_address.split('<').last.split('>').first
```

Comment:
The mail gem already handles friendly display names, so this is useless. the only logic that needs to be done is if there is a `@parse_error` when `allow_display_name` is set and `address.address` is present.